### PR TITLE
Navigation child test pages

### PR DESCRIPTION
**Pull request by wachterjohannes**
__tasks:__

- [x] test coverage
- [ ] gather feedback for my changes

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | fixes sulu-io/sulu-standard#311
| BC breaks        | sub pages of test-pages will not be merge into upper layer
| Documentation PR | none

### CODE REVIEW

**Comment by wachterjohannes** [comment]
@chirimoya we have to talk about the html sitemap for this case! @crash0verride says they should be displayed

**Comment by wachterjohannes** [comment]
i have talked to @chirimoya and we said that we remove children of testpages (and pages not in the context) but in the sitemap we keep the old behauviour

File: UPGRADE.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Upgrade
 
+## dev-develop
+
+### Website Navigation
+
+Children of Test-Pages wont be merged into the upper layer. They will be ignored in the tree navigation.
```
Comment (danrot) [inline UPGRADE.md]: We probably should also mention the navigation property here.

File: src/Sulu/Component/Content/Query/ListToTreeConverter.php [hunk 2]
```diff
@@ -83,7 +97,10 @@ private function toArray($tree)
             // search for empty nodes
             for ($i = 0; $i < sizeof($tree['children']); $i++) {
                 if (array_keys($tree['children'][$i]) === array('children')) {
-                    array_splice($tree['children'], $i + 1, 0, $tree['children'][$i]['children']);
+                    if(!!$this->moveUp) {
```
Comment (danrot) [inline src/Sulu/Component/Content/Query/ListToTreeConverter.php]: We don't do `!!` in php :wink:

**Comment by wachterjohannes** [comment]
fixed

**Pull request merged**